src/main/java/jung/gui/DialogBox.java

```diff
@@ -0,0 +1,61 @@
+package jung.gui;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+/**
+ * Represents a dialog box consisting of an ImageView to represent the speaker's face
+ * and a label containing text from the speaker.
+ */
+public class DialogBox extends HBox {
+    @FXML
+    private Label dialog;
+    @FXML
+    private ImageView displayPicture;
+
+    private DialogBox(String text, Image img) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBox.fxml"));
+            fxmlLoader.setController(this);
+            fxmlLoader.setRoot(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        dialog.setText(text);
+        displayPicture.setImage(img);
+    }
+
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
+        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
+        Collections.reverse(tmp);
+        getChildren().setAll(tmp);
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
+    public static DialogBox getJungDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
+        db.flip();
+        return db;
+    }
```

Comment:
The main area for improvement is error handling and visual styling differentiation. The current flip() method only changes the layout order.
```suggestion
/**
 * Flips the dialog box such that the ImageView is on the left and text on the right.
 * Also applies a different style to differentiate Jung's messages from the user's.
 */
private void flip() {
    ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
    Collections.reverse(tmp);
    getChildren().setAll(tmp);
    setAlignment(Pos.TOP_LEFT);

    // Add a style class to Jung's dialog for distinct styling (e.g., background color)
    this.getStyleClass().add("jung-dialog");
}

public static DialogBox getUserDialog(String text, Image img) {
    DialogBox db = new DialogBox(text, img);
    // Add a style class to the user's dialog for distinct styling
    db.getStyleClass().add("user-dialog");
    return db;
}

public static DialogBox getJungDialog(String text, Image img) {
    var db = new DialogBox(text, img);
    db.flip(); // This now also adds the 'jung-dialog' class
    return db;
}
```